my-app/src/ListIterator.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class ListIterator extends Component {
+  constructor(props) {
+    super(props);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  handleDelete(event) {
+    let clickedButtonElement = event.currentTarget.previousElementSibling.textContent;
+    this.props.deleteItem(clickedButtonElement);
+  }
+
+  render() {
+    return (
+      <ul>
+        {
+          this.props.listItems.map((item, index) => {
+            return <li key={index}>
+                    <span>{item}</span>
+                    <button onClick={this.handleDelete}>x</button>
+                  </li>
```

Comment:
Start writing properly indented code. It makes the code more readable and sensible.